lib/ssh_scan/config_loader/yaml.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+module SSHScan
+  module ConfigLoader
+    class Yaml
+      def initialize(config, class_name = nil, mode = 'file', logger = nil)
+        @opts = case mode
+          when "file" then YAML.load_file(config)
+          when "string" then YAML.load(config)
+        end
```

Comment:
`ConfigLoader` won't require any of this, now that we're not having mode at all, since it can be guessed. 

I think it is okay to have this in internal files like `yaml.rb` that do this processing. I agree they are sort of "dumb", but the combined use makes `ConfigLoader` smart.